logstage/logstage-sink-file/src/main/scala/com/github/pshirshov/izumi/FileSink.scala

```diff
@@ -0,0 +1,88 @@
+package com.github.pshirshov.izumi
+
+import java.util.concurrent.atomic.AtomicReference
+import com.github.pshirshov.izumi.models._
+import com.github.pshirshov.izumi.logstage.api.logger.RenderingPolicy
+import com.github.pshirshov.izumi.logstage.model.Log
+import com.github.pshirshov.izumi.logstage.model.logger.LogSink
+import com.github.pshirshov.izumi.logstage.sink.console.ConsoleSink
+import scala.util.{Failure, Success, Try}
+
+
+class FileSink(policy: RenderingPolicy, consoleSink: ConsoleSink, config: FileSinkConfig) extends LogSink {
+
+  private val sinkState = new AtomicReference[SinkState](SinkState(config = config))
```

Comment:
The contents of SinkState are mutable, AtomicReference doesn't add thread safety here.